pkg/common/quotas/quotas.go

```diff
@@ -148,39 +153,60 @@ func (q *QuotaChecker) getNamespaceOverrides(ctx context.Context, c client.Clien
 
 // AddInitalNamespaceQuotas will add the default quotas to the namespace annotations.
 // It will only add them, if there are currently no such annotations in place.
-func AddInitalNamespaceQuotas(ns *corev1.Namespace) bool {
+func AddInitalNamespaceQuotas(ctx context.Context, iof *iofruntime.Runtime, ns *corev1.Namespace, objMeta *metaonly.MetadataOnlyObject) bool {
 	annotations := ns.GetAnnotations()
 	if annotations == nil {
 		annotations = map[string]string{}
 	}
 
 	added := false
 
+	r := utils.GetDefaultResources()
+
+	addSidecarResources(ctx, objMeta, iof, r)
+
 	if _, ok := annotations[utils.DiskAnnotation]; !ok {
 		annotations[utils.DiskAnnotation] = utils.DefaultDiskRequests.String()
 		added = true
 	}
 
 	if _, ok := annotations[utils.CpuRequestAnnotation]; !ok {
-		annotations[utils.CpuRequestAnnotation] = utils.DefaultCPURequests.String()
+		annotations[utils.CpuRequestAnnotation] = r.CPURequests.String()
 		added = true
 	}
 
 	if _, ok := annotations[utils.CpuLimitAnnotation]; !ok {
-		annotations[utils.CpuLimitAnnotation] = utils.DefaultCPULimit.String()
+		annotations[utils.CpuLimitAnnotation] = r.CPULimits.String()
 		added = true
 	}
 
 	if _, ok := annotations[utils.MemoryRequestAnnotation]; !ok {
-		annotations[utils.MemoryRequestAnnotation] = utils.DefaultMemoryRequests.String()
+		annotations[utils.MemoryRequestAnnotation] = r.MemoryRequests.String()
 		added = true
 	}
 
 	if _, ok := annotations[utils.MemoryLimitAnnotation]; !ok {
-		annotations[utils.MemoryLimitAnnotation] = utils.DefaultMemoryLimits.String()
+		annotations[utils.MemoryLimitAnnotation] = r.MemoryLimits.String()
 		added = true
 	}
 
 	ns.SetAnnotations(annotations)
 	return added
 }
+
+// addSidecarResources will add the sidecar resources to the given resources.
+func addSidecarResources(ctx context.Context, objMeta *metaonly.MetadataOnlyObject, iof *iofruntime.Runtime, r *utils.Resources) {
+	l := ctrl.LoggerFrom(ctx)
+	if objMeta.Kind == "XVSHNPostgreSQL" {
+		comp := vshnv1.VSHNPostgreSQL{}
+		err := iof.Desired.GetComposite(ctx, &comp)
+		var instances int
+		if err != nil {
+			l.Info("Could not get composite. Assuming 1 instance.")
```

Comment:
why do we assume? what are the situations when we can't access Composite?

Reply:
This has been removed in some refactorings